examples/yacat/yacat.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python3
+import asyncio
+import pathlib
+import sys
+
+from yapapi.log import enable_default_logger, log_summary, log_event_repr  # noqa
+from yapapi.runner import Engine, Task, vm
+from yapapi.runner.ctx import WorkContext
+from datetime import timedelta
+
+# For importing `utils.py`:
+script_dir = pathlib.Path(__file__).resolve().parent
+parent_directory = script_dir.parent
+sys.stderr.write(f"Adding {parent_directory} to sys.path.\n")
+sys.path.append(str(parent_directory))
+import utils  # noqa
+
+
+def write_hash(hash):
+    f = open(str(script_dir / "in.hash"), "w")
+    f.write(hash)
+    f.close()
+
+
+def write_keyspace_check_script(mask):
+    f = open(str(script_dir / "keyspace.sh"), "w")
+    f.write(f"hashcat --keyspace -a 3 {mask} -m 400 > /golem/work/keyspace.txt")
+    f.close()
+
+
+def read_keyspace():
+    f = open(str(script_dir / "keyspace.txt"), "r")
```

Comment:
This will only work if you run the example from the script's directory. In general, `keyspace.txt` will be written to the current working directory, which does not need to be `script_dir`. 

You need to use just `"keyspace.txt"`, without the `script_dir` prefix.

Reply:
corrected